### PR TITLE
IP Whitelist: Fix IPv4/IPv6 parsing for empty string

### LINKED ISSUE #1833: Adding only IPv6 address to registrar ip whitelist will result in empty string for IPv4 address
empty sting in ipv4 address list will result with inability to sign in nor sign out from the portal with internal error.

Logs: IPAddr::AddressFamilyError (address family mismatch)

The IPv4 or IPv6 value for any record if missing must be NULL.

### PR DESCRIPTION
**Pull request by OlegPhenomenon**
Closes #1833

### CODE REVIEW

File: test/models/white_ip_test.rb [hunk 1]
```diff
@@ -38,6 +38,16 @@ def test_validates_ipv6_format
     assert white_ip.valid?
   end
 
+  def test_validates_include_empty_ipv4
+    white_ip = valid_white_ip
+
+    white_ip.update(ipv4: '')
```
Comment (yulgolem) [inline test/models/white_ip_test.rb:44]: Not sure 100%, but as far as I remember `#update` method may bypass model validations.
I'd suggest to:
1. Create WhiteIp model (`white_ip = WhiteIp.new`)
2. assign `white_ip.ipv6`.
3. Assert no exception risen on saving `white_ip`
4. assert `white_ip.valid?`

**Comment by yulgolem** [comment]
BTW maybe we should add a system test for that?
With filling fields, pressing buttons and such?

**Review by yulgolem — APPROVED** [review]

**Review by yulgolem — APPROVED** [review]

**Comment by vohmar** [comment]
The empty ipv4 column is still filled with empty string ('') - no improvement

testing steps:
1. added new whitelisted ip record to a registrar (eedirect) in admin (/admin/registrars/2/white_ips/new) - no ipv4, ipv6 2001:7d0:83b6:5f84::27f
image
2. tried to access registrar portal - internal error
3. select	ipv4
  from	white_ips
  where	ipv6 = '2001:7d0:83b6:5f84::27f';

result - empty string
4. update	white_ips
  set	ipv4 = NULL
  where ipv6 = '2001:7d0:83b6:5f84::27f';

all works fine

**Comment by yulgolem** [comment]
Suppose the solution need to be not on the saving/validating side, but on the WhiteIp checking side.
I'll take it.

**Pull request merged**